collectors/ebpf.plugin/ebpf_softirq.c

```diff
@@ -0,0 +1,274 @@
+    {.name = "BLOCK", .latency = 0},
+    {.name = "IRQ_POLL", .latency = 0},
+    {.name = "TASKLET", .latency = 0},
+    {.name = "SCHED", .latency = 0},
+    {.name = "HRTIMER", .latency = 0},
+    {.name = "RCU", .latency = 0},
+};
+
+// tmp store for soft IRQ values we get from a per-CPU eBPF map.
+static softirq_ebpf_val_t *softirq_ebpf_vals = NULL;
+
+static struct bpf_link **probe_links = NULL;
+static struct bpf_object *objects = NULL;
+
+static int read_thread_closed = 1;
+
+static struct netdata_static_thread softirq_threads = {"SOFTIRQ KERNEL",
+                                                    NULL, NULL, 1, NULL,
+                                                    NULL, NULL };
+
+/**
+ * Clean up the main thread.
+ *
+ * @param ptr thread data.
+ */
+static void softirq_cleanup(void *ptr)
+{
+    for (int i = 0; softirq_tracepoints[i].class != NULL; i++) {
+        ebpf_disable_tracepoint(&softirq_tracepoints[i]);
+    }
+
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+    if (!em->enabled) {
+        return;
+    }
+
+    heartbeat_t hb;
+    heartbeat_init(&hb);
+    uint32_t tick = 1 * USEC_PER_MS;
+    while (!read_thread_closed) {
+        usec_t dt = heartbeat_next(&hb, tick);
+        UNUSED(dt);
+    }
+
+    freez(softirq_ebpf_vals);
+    freez(softirq_threads.thread);
+
+    if (probe_links) {
+        struct bpf_program *prog;
+        size_t i = 0 ;
+        bpf_object__for_each_program(prog, objects) {
+            bpf_link__destroy(probe_links[i]);
+            i++;
+        }
+        bpf_object__close(objects);
+    }
+}
+
+/*****************************************************************
+ *  MAIN LOOP
+ *****************************************************************/
+
+static void softirq_read_latency_map()
+{
+    int fd = softirq_maps[SOFTIRQ_MAP_LATENCY].map_fd;
+    int i;
+    for (i = 0; i < NETDATA_SOFTIRQ_MAX_IRQS; i++) {
+        int test = bpf_map_lookup_elem(fd, &i, softirq_ebpf_vals);
+        if (unlikely(test < 0)) {
+            continue;
+        }
+
+        uint64_t total_latency = 0;
+        int cpu_i;
+        int end = ebpf_nprocs;
+        for (cpu_i = 0; cpu_i < end; cpu_i++) {
+            total_latency += softirq_ebpf_vals[cpu_i].latency/1000;
+        }
+
+        softirq_vals[i].latency = total_latency;
+    }
+}
+
+/**
+ * Read eBPF maps for soft IRQ.
+ */
+static void *softirq_reader(void *ptr)
+{
+    read_thread_closed = 0;
+
+    heartbeat_t hb;
+    heartbeat_init(&hb);
+
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+
+    usec_t step = NETDATA_SOFTIRQ_SLEEP_MS * em->update_time;
+    while (!close_ebpf_plugin) {
+        usec_t dt = heartbeat_next(&hb, step);
+        UNUSED(dt);
+
+        softirq_read_latency_map();
+    }
+
+    read_thread_closed = 1;
+    return NULL;
+}
+
+static void softirq_create_charts()
+{
+    ebpf_create_chart(
+        NETDATA_EBPF_SYSTEM_GROUP,
+        "softirq_latency",
+        "Software IRQ latency",
+        "milliseconds",
```

Comment:
We have a list of common units [here](https://github.com/netdata/netdata/blob/master/collectors/ebpf.plugin/ebpf.h#L219-L225), considering this is an "unit" used with `hard` and `soft` IRQ, I will suggest you to create a preprocessor for it.

Reply:
Addressed in https://github.com/netdata/netdata/pull/11445/commits/ba9cbddce1ebb91fccbc059a8fbdd7cc0a386424